### DISCUSSION TITLE
Unable to switch default_branch of a repository using GitHub Apps Installation Token

### DISCUSSION THREAD

**Discussion by jjayabal23** [community/community#62371] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

I have a GitHub App with the following Repository Permission,
`Administration` >> `Read & Write`

I have generated a GitHub App installation token using octokit.net and tried calling the PATCH /repos/owner/repo to switch the default_branch to master, but the API returns the following error with status code 422 although the master branch already exists,

**Request:**
```
{
	"default_branch": "master"
}
```
**Response:**
```
{
	"message": "Validation Failed",
	"errors": [
		{
			"message": "The branch master was not found. Please push that ref first or create it via the Git Data API.",
			"resource": "Repository",
			"field": "default_branch",
			"code": "invalid"
		}
	],
	"documentation_url": "https://docs.github.com/rest/repos/repos#update-a-repository"
}
```
The same API works fine when I use a PAT of a User / Collaborator. Switching default_branch was working until 28/07/2023 and started throwing the above error thereafter.

**Expected:**
GitHub App should have the capapbility to switch the default_branch of a repository for which the Administrator Read & Write permissions for the repository is enabled.

**Comment by ghost** [comment]
Did you ever find a solution for this @jjayabal23? I've just run into the same issue, I even have contents write permission and can can lookup the new branch name via https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28#get-a-reference fine but the change default branch request still fails.

I also run into a similar issue if I try to rename the default branch which the docs say is possible.

**Comment by ghost** [comment]
I've figured it out, it seems you need both repository admin as well as org admin permissions on the GitHub App, just one or the other fails with this error. Smells like someone stuffed up and included the wrong admin ACL check somewhere along the line as I can't imagine why this would require org admin rights to change the default branch of a repo and the docs make no mention of this either.

**Comment by alexwoodrow-dimension** [comment]
I am also encountering this issue

**Reply by dev-james-ev** [reply]
@alexwoodrow-dimension are you using org level rulesets? https://docs.github.com/en/enterprise-cloud@latest/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch#prerequisites

It's pretty bad that info is only in the enterprise server level docs rather then the team level docs if it applies to both

**Reply by alexwoodrow-dimension** [reply]
You are entirely correct. We sat down yesterday and tried a few things and it turned out enabling any rule in the org level ruleset disabled repo admin ability to change the default branch. Made sense to us after a chat - being able to change the default branch might allow a repo admin to bypass an org rule.

But yes, it would be really helpful for the text in the Enterprise Cloud docs

https://docs.github.com/en/enterprise-cloud@latest/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch#prerequisites

"Rulesets at the organization or enterprise level that apply to branches of a repository will not allow the repository administrator to rename branches of the targeted repository or change the default branch to another branch."

to also be in the Free/Pro/Teams docs

https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch#prerequisites
